### PR TITLE
Add output if a successful authentication is via Guest privileges

### PR DESCRIPTION
**Pull request by Marshall-Hallenbeck**
@Dfte wrote a great post about Guest vs NULL sessions [here](https://sensepost.com/blog/2024/guest-vs-null-session-on-windows/), so I have updated the SMB plaintext&hash login code to tell the user if the login was via Guest privileges.

Hopefully this along with the blog post helps alleviate confusion!

@mpgn @NeffIsBack I added code to check if the hash is 16 or 32 bytes, do you think it's necessary? Helps catch if people off-by-one the length of hashes. There's no reason the hash _should_ be anything else, right?

image

### CODE REVIEW

**Comment by Dfte** [comment]
Love it ahah! Yeah I see no reason why hashes could be anything except 16/32 :)

**Comment by mpgn** [comment]
yeah nice :)

**Review by mpgn — DISMISSED** [review]

File: nxc/connection.py [hunk 1]
```diff
@@ -554,7 +563,7 @@ def login(self):
                         return True
 
     def mark_pwned(self):
-        return highlight(f"({pwned_label})" if self.admin_privs else "")
+        return highlight(f" ({pwned_label})" if self.admin_privs else "")
```
Comment (NeffIsBack) [inline nxc/connection.py:566]: If you add the space here you probably have to change all other protocols, which use the function. Imo we should just leave it as it is, as the string where it is placed into should have the formatting anyways.

Comment (Marshall-Hallenbeck) [reply nxc/connection.py:566]: Fair enough

**Comment by Marshall-Hallenbeck** [comment]
@NeffIsBack uhh weird, that is a package error, and I didn't change anything about the packages.

**Review by NeffIsBack — APPROVED** [review]
LGTM
image

**Comment by NeffIsBack** [comment]
Oh, @Marshall-Hallenbeck you did add a hash file but not a test using it, did you?

**Comment by Marshall-Hallenbeck** [comment]
@NeffIsBack oop yeah you're right, I'll add 'em and PR real quick

**Pull request merged**